app/controllers/summoners_controller.rb

```diff
@@ -50,4 +49,113 @@ def find_by_name
       end
     end
   end
+
+  def league_entry
+    require 'open-uri'
+    require 'json'
+
+    league = {
+      solo5: nil,
+      team5: nil,
+      team3: nil
+    }
+
+    @tries = 0
+    begin
+      url = "https://#{params['region']}.api.pvp.net/api/lol/#{params['region'].downcase}/v2.5/league/by-summoner/#{params['id']}/entry?api_key=#{ENV['API_KEY']}"
+      response = JSON.parse(open(url).read)
+      if response.empty?
+        render json: {success: true, league_entry: nil}, status: 200
+      else
```

Comment:
Seems like the code is very "imperative", meaning not very modular. You can try modularizing it if you have time